### ISSUE TITLE
Answer Relevance doesnt need context

### ISSUE THREAD

**Issue by mohamad-tohidi** [explodinggradients/ragas#1177]
[ ] I have checked the [documentation](https://docs.ragas.io/) and related resources and couldn't resolve my bug.

**Describe the bug**
the answer relevance metric, does not need context as input, it doesnt use it anywhere
so why is it when i try to run it without contexts list it gives me error?

even in the documentation you mentioned that it needs contexts, but you did not used it anywhere

**Comment by knpunk** [comment]
I'd like to add to this that, nowhere in the prompt it's instructing to go through the contexts. It's just taking care of non-committals and for contexts it's just few shot prompting. Maybe due to this I've noticed it's giving better scores for some of the wrong answers as compared to the correct ones. I'm no expert to say this but I think the prompt needs a little more fine tuning.

Cheers and thanks!

**Comment by knpunk** [comment]
@dosu I'd like to add to this that, nowhere in the prompt it's instructing to go through the contexts. It's just taking care of non-committals and for contexts it's just few shot prompting. Maybe due to this I've noticed it's giving better scores for some of the wrong answers as compared to the correct ones. I'm no expert to say this but I think the prompt needs a little more fine tuning.

Cheers and thanks!

**Comment by gustavocidornelas** [comment]
Looking at the prompt, I also feel like the contexts shouldn't be required for answer relevance.

I understand the motivation for adding it in, exposed by @shahules786 on issue https://github.com/explodinggradients/ragas/issues/440, but for many other cases, the context is not critical.

  @jjmachan, if that's ok, I could prepare a PR that makes the `context` optional: if it's provided, fine, maintain the current behavior; otherwise, generate the questions based on the answer alone. What do you think?

**Comment by jjmachan** [comment]
@gustavocidornelas your help would be highly appreciated - thanks a lot for offering! @shahules786 what do you think should be the next step?

**Comment by shahules786** [comment]
You're right. From the past experience, it seems to me that adding contexts just to fit edge cases in the relevancy metric is not very useful. Just raised a PR to fix this issue.
